community/lucene-index/src/main/java/org/neo4j/index/impl/lucene/explicit/LuceneDataSource.java

```diff
@@ -298,7 +298,7 @@ IndexReference getIndexSearcher( IndexIdentifier identifier ) throws ExplicitInd
         {
             return syncGetIndexSearcher( identifier );
         }
-        synchronized ( searcher )
+        synchronized ( this )
```

Comment:
It looks like the path into this `synchronized` block is the most common one. Won't this change increase contention a bit too much?

Reply:
@tinwelint I don't see any other simple way around it. This class is pretty much the all-star team of concurrency control with `synchronized` blocks both on self and inner instances that themselves also have `synchronized` methods, one concurrent map, two additional concurrent collections hidden inside `IndexClockCache` and then also a `ReentrantReadWriteLock` that can be grabbed from outside.

Untangling this seems like a bigger risk to me, right now.